eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -82,12 +82,14 @@ case class Features(activated: Map[Feature, FeatureSupport], unknown: Set[Unknow
     unknownFeaturesOk && knownFeaturesOk
   }
 
-  def initFeatures(): Features = Features(activated.collect { case (f: InitFeature, s) => (f: Feature, s) }, unknown)
+  def initFeatures(): Features[InitFeature] = Features[InitFeature](activated.collect { case (f: InitFeature, s) => (f, s) }, unknown)
 
-  def nodeAnnouncementFeatures(): Features = Features(activated.collect { case (f: NodeFeature, s) => (f: Feature, s) }, unknown)
+  def nodeAnnouncementFeatures(): Features[NodeFeature] = Features[NodeFeature](activated.collect { case (f: NodeFeature, s) => (f, s) }, unknown)
 
-  // NB: we don't include unknown features in invoices, which means plugins cannot inject invoice features.
-  def invoiceFeatures(): Map[Feature with InvoiceFeature, FeatureSupport] = activated.collect { case (f: InvoiceFeature, s) => (f, s) }
+  // NB: features above 5 * 2^10 do not fit in bolt 11 invoices
+  def invoiceFeatures(): Features[InvoiceFeature] = Features[InvoiceFeature](activated.collect { case (f: InvoiceFeature, s) => (f, s) }, unknown.filter(_.bitIndex < 5 * 1024))
```

Comment:
There was a lot of duplication from `Features` functions inside `PaymentRequest` so I wanted to use `Features` everywhere. I was planning to add better typing later but since you asked, I've added it with fd6b9d0e754a934c110df149b5de63d86093b930

Reply:
Please don't add this in this PR, this is really unrelated to the refactoring of payment request / invoice.
We'd discussed this change [in a previous PR](https://github.com/ACINQ/eclair/pull/2063#discussion_r779601182) and it was left out because it's not so clear-cut that it's a win, so it really deserves to be in its own PR.

Reply:
Sorry I realize that my comment "now is a good time" was misleading, I meant once this PR is integrated, we could tackle the feature types. My bad.

Reply:
What changes exactly do you think should be in a separate PR?

Reply:
I'd say almost all of https://github.com/ACINQ/eclair/commit/fd6b9d0e754a934c110df149b5de63d86093b930 (making the `Features` class generic), I'd like to have this properly reviewed and discussed on its own.

Reply:
I've reverted it

Reply:
:pray: